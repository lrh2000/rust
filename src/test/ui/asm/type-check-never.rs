// only-x86_64

#![feature(asm, never_type)]
#[allow(dead_code)]

fn main() {
    unsafe {
        // Type checks ignore never type
        // But type annotations are needed
        // See also type-check-2.rs

        let u: ! = unreachable!();
        asm!("{}", in(reg) u); //~ ERROR: type annotations needed
    }
}
