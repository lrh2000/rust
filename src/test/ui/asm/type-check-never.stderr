error[E0282]: type annotations needed
  --> $DIR/type-check-never.rs:13:28
   |
LL |         asm!("{}", in(reg) u);
   |                            ^ cannot infer type
   |
   = note: type must be known at this point

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
